safe/impact_statistics/aggregator.py

```diff
@@ -171,6 +171,23 @@ def __init__(self, extent, aggregation_layer):
             self.write_keywords(
                 self.layer, keywords)
 
+    @property
+    def extent(self):
+        return self._extent
+
+    @extent.setter
+    def extent(self, value):
+        self._extent = value
+        # update layer extent to match impact layer if in aoi_mode
+        if self.aoi_mode:
+            try:
+                self.layer = self._extents_to_layer()
+                self.safe_layer = safe_read_layer(str(self.layer.source()))
```

Comment:
Remove casting it to str. It will raise unicode error if the layer source's got some non ASCII